From 03de9d9434986bc36cb0fef3367ce8e53549624e Mon Sep 17 00:00:00 2001
From: Zhantao Tang <zhantao.tang@windriver.com>
Date: Tue, 16 Aug 2022 15:24:38 +0800
Subject: [PATCH] pfe: netif: use the correct ndo_eth_ioctl to fix the ptp
 unsupport issue

This patch is to fix the following error:

  # ptp4l -A -4 -H -m -i pfe2
  ptp4l[1601.644]: selected /dev/ptp2 as PTP clock
  ptp4l[1601.661]: driver rejected most general HWTSTAMP filter
  ptp4l[1601.661]: ioctl SIOCSHWTSTAMP failed: Operation not supported
  ptp4l[1601.665]: port 1: INITIALIZING to FAULTY on FAULT_DETECTED (FT_UNSPECIFIED)
  ptp4l[1601.668]: port 0: INITIALIZING to LISTENING on INIT_COMPLETE

i.e. the hardware timestamp is not supported in pfe driver.

The reason is that, in kernel commit a76053707db ("dev_ioctl: split out
ndo_eth_ioctl"), the original ndo_do_ioctl is replace by ndo_eth_ioctl
to implement the commands, like hardware timestamping:
	SIOCSHWTSTAMP/SIOCGHWTSTAMP

So update pfe driver to use the correct ndo_eth_ioctl interface can fix
the above issue.

Upstream-Status: Pending

Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 sw/linux-pfeng/pfeng-netif.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sw/linux-pfeng/pfeng-netif.c b/sw/linux-pfeng/pfeng-netif.c
index 93c29f9..faec6ea 100644
--- a/sw/linux-pfeng/pfeng-netif.c
+++ b/sw/linux-pfeng/pfeng-netif.c
@@ -642,7 +642,7 @@ static const struct net_device_ops pfeng_netdev_ops = {
 	.ndo_start_xmit		= pfeng_netif_logif_xmit,
 	.ndo_stop		= pfeng_netif_logif_stop,
 	.ndo_change_mtu		= pfeng_netif_logif_change_mtu,
-	.ndo_do_ioctl		= pfeng_netif_logif_ioctl,
+	.ndo_eth_ioctl		= pfeng_netif_logif_ioctl,
 	.ndo_set_mac_address	= pfeng_netif_set_mac_address,
 	.ndo_set_rx_mode	= pfeng_netif_set_rx_mode,
 	.ndo_fix_features	= pfeng_netif_fix_features,
-- 
2.25.1

